### PR TITLE
color-identifiers-mode.el: simplify while loop a bit

### PR DESCRIPTION
**Pull request by Hi-Angel**
Instead of throwing and catching an exception let's just pre-assign the
variable. It take less code, less indentation, and is faster.

### CODE REVIEW

File: color-identifiers-mode.el [hunk 1]
```diff
@@ -174,18 +174,17 @@ For cc-mode support within color-identifiers-mode."
...
+        (while next-change
+          (goto-char next-change)
+          (when (or (eq (get-text-property (point) 'face) 'font-lock-variable-name-face)
+                    ;; If we fontified it in the past, assume it should
+                    ;; continue to be fontified. This avoids alternating
+                    ;; between fontified and unfontified.
+                    (get-text-property (point) 'color-identifiers:fontified))
+            (push (substring-no-properties (symbol-name (symbol-at-point))) result))
+          (setq next-change (next-property-change (point)))
+          )))
```
Comment (ankurdave) [inline color-identifiers-mode.el:187]: I think Lisp style is to keep the closing parens with the previous line.

```suggestion
          (setq next-change (next-property-change (point))))))
```

File: color-identifiers-mode.el [hunk 2]
```diff
@@ -310,18 +309,17 @@ arguments, loops (for .. in), or for comprehensions."
...
+          (while next-change
+            (goto-char next-change)
+            (when (or (eq (get-text-property (point) 'face) 'font-lock-variable-name-face)
+                      ;; If we fontified it in the past, assume it should
+                      ;; continue to be fontified. This avoids alternating
+                      ;; between fontified and unfontified.
+                      (get-text-property (point) 'color-identifiers:fontified))
+              (push (substring-no-properties (symbol-name (symbol-at-point))) result))
+            (setq next-change (next-property-change (point)))
+            )))
```
Comment (ankurdave) [inline color-identifiers-mode.el:322]:
```suggestion
            (setq next-change (next-property-change (point))))))
```

File: color-identifiers-mode.el (hunk 1 above)
Comment (Hi-Angel) [reply color-identifiers-mode.el]: Thanks! Done!

File: color-identifiers-mode.el (hunk 2 above)
Comment (Hi-Angel) [reply color-identifiers-mode.el]: Thanks! Done!

**Comment by ankurdave** [comment]
Thanks!

**Pull request merged**